### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by xylar**
* Add more detail on typical usage
* Add installation instructions

### CODE REVIEW

**Comment by xylar** [comment]
@forsyth2, could you have a look at the modified README:
https://github.com/E3SM-Project/mache/blob/42b98eccdbf5c74b3b595e156f8f74b5f99c60ed/README.rst
and let me know if it clarifies questions you had in https://github.com/E3SM-Project/zppy/pull/237?

File: README.rst [hunk 1]
```diff
@@ -9,12 +9,18 @@ Example usage:
 .. code-block:: python
 
     #!/usr/bin/env python
-    from mache import MachineInfo
+    from mache import MachineInfo, discover_machine
 
-    machinfo = MachineInfo(machine='anvil')
+    machinfo = MachineInfo()
     print(machinfo)
+    diags_base = machine_info.config.get('diagnostics', 'base_path')
```
Comment (forsyth2) [inline README.rst:16]: `machine_info` should match with `machinfo` two lines above.

**Review by forsyth2 — APPROVED** [review]
@xylar Thanks, yes, this clarifies both how to extract a path from `MachineInfo` and how to create an environment with Mache.  

Also, re: "cori-haswell (but you will get a warning)": ok, I see `UserWarning: defaulting to cori-haswell.  Use -m cori-knl if you wish to run on KNL.` It might be clearer to just print that warning in the documentation or specify "warning that `knl` is also an option"

**Comment by xylar** [comment]
@forsyth2, thank you for the feedback.  I addressed your correction and suggestion.

**Pull request merged**